### PR TITLE
chore: add java package to protos

### PR DESCRIPTION
**Pull request by eddiezane**
# Pull Request

## What? (description)

This PR adds `java_package` options to all protos.

## Why? (reasoning)

I am generating a Java/Kotlin library to interact with clusters from an Android app. Java packages, much like Go, must match their folder path on disk. You cannot provide `java_package` as a build time option to the Java protoc compiler. So for the package to be defined as anything other than `cluster`, the option must be defined at the proto level. I set it to what I thought was reasonable but totally open to changing it.

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

File: api/resource/definitions/cluster/cluster.proto [hunk 1]
```diff
@@ -3,6 +3,7 @@ syntax = "proto3";
 package talos.resource.definitions.cluster;
 
 option go_package = "github.com/siderolabs/talos/pkg/machinery/api/resource/definitions/cluster";
+option java_package = "dev.talos.api.resource.definitions.cluster";
```
Comment (smira) [inline api/resource/definitions/cluster/cluster.proto:6]: the files under `api/resource/definitions` are automatically generated (you can check that by running `make generate`), so you'd need to update the generator (should be hack/structprotogen iirc)

Comment (eddiezane) [reply api/resource/definitions/cluster/cluster.proto:6]: Added and generated!

**Review by smira — APPROVED** [review]
thank you!

**Pull request merged**